src/edge/edgeManager.ts

```diff
@@ -404,16 +408,24 @@ export class EdgeManager {
         // TODO command to create module;
         switch (template) {
             case Constants.LANGUAGE_CSHARP:
-                await Executor.executeCMD(outputChannel, "dotnet", { shell: true }, "new -i Microsoft.Azure.IoT.Edge.Module");
+                if (Versions.installCSharpTemplate()) {
+                    const csversion = Versions.csTemplateVersion();
+                    const installCmd = csversion != null ? `new -i Microsoft.Azure.IoT.Edge.Module::${csversion}` : "new -i Microsoft.Azure.IoT.Edge.Module";
+                    await Executor.executeCMD(outputChannel, "dotnet", { shell: true }, installCmd);
+                }
                 await Executor.executeCMD(outputChannel, "dotnet", { cwd: `${parent}`, shell: true }, `new aziotedgemodule -n "${name}" -r ${repositoryName}`);
                 break;
             case Constants.CSHARP_FUNCTION:
-                await Executor.executeCMD(outputChannel, "dotnet", { shell: true }, "new -i Microsoft.Azure.IoT.Edge.Function");
+                if (Versions.installCSFunctionTemplate()) {
+                    const csfuncversion = Versions.csFunctionTemplateVersion();
+                    const installCmd = csfuncversion != null ? `new -i Microsoft.Azure.IoT.Edge.Function::${csfuncversion}` : "new -i Microsoft.Azure.IoT.Edge.Function";
+                    await Executor.executeCMD(outputChannel, "dotnet", { shell: true }, installCmd);
+                }
                 await Executor.executeCMD(outputChannel, "dotnet", { cwd: `${parent}`, shell: true }, `new aziotedgefunction -n "${name}" -r ${repositoryName}`);
                 break;
             case Constants.LANGUAGE_PYTHON:
                 const gitHubSource = "https://github.com/Azure/cookiecutter-azure-iot-edge-module";
-                const branch = "master";
+                const branch = Versions.pythonTemplateVersion();

```

Comment:
The default branch of https://github.com/Azure/cookiecutter-azure-iot-edge-module is develop, so if branch is undefined, users will get develop template.